demo/src/task-scheduler.ts

```diff
@@ -0,0 +1,40 @@
+// Faster setTimeout(fn, 0) implementation using postMessage API
+// Based on https://dbaron.org/log/20100309-faster-timeouts
+export class MicroTaskScheduler {
+  readonly messageName = 'zero-timeout-message';
+
+  private executionQueue: Function[];
+  private stopped: boolean;
+
+  constructor() {
+    this.executionQueue = new Array<Function>();
+    this.stopped = true;
+  }
+
+  start() {
+    this.stopped = false;
```

Comment:
I'd add a check if we are stopped here, to make sure we never add the event handler twice (in case of two consecutive calls to `start()`)